### PR TITLE
Idiomatic "clap" usage (implements --version switch for tools)

### PR DESCRIPTION
**Pull request by villesundell**
## Motivation

I wanted to provide `--version` switches for some of the tools. However, I noticed that almost none of the tools implemented that switch. Also, "clap" usage was inconsistent across the tools. So, ended up updating Cargo.toml for each of the tools, and simplified the "clap" usage for these tools.

When Move returns to versioned releases, the tools are now ready to communicate that information to the user :+1: 

The main commit:
```
[language] Idiomatic "clap" usage in CLI tools

Following the "clap" documentation on basic "clap" usage (fetching
the information from Cargo.toml). Now "clap" usage is (mostly)
consistent across the tools, the only exceptions being
move-compiler's move-check tool, and move-coverage's
coverage-summaries, move-trace-conversion and source-coverage. These
tools do not have their own Cargo.tomls.

Followed example:
github.com/clap-rs/clap/blob/v3.1.7/examples/cargo-example-derive.rs

Related commit (updating Cargo.toml descriptions for these tools):
096562e61 ("[language] Updated descriptions for some tools",
2022-04-26).

```

I followed this [example](github.com/clap-rs/clap/blob/v3.1.7/examples/cargo-example-derive.rs) on clap `derive` usage.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/move-language/move/blob/main/CONTRIBUTING.md#pull-requests)?

Yes. Still the anchor `#pull-requests` is missing.

## Test Plan

Went manually through each binary, and executed `-V` switch on them. Only CLI tools not implementing the `-V` switch were non-clap-based `prover-lab`, `move-stdlib` and `prover-mutation`.

### CODE REVIEW

File: language/move-compiler/src/bin/move-build.rs [hunk 1]
```diff
@@ -11,7 +11,7 @@ use move_compiler::{
 };
 
 #[derive(Debug, Parser)]
-#[clap(name = "Move Build", about = "Compile Move source to Move bytecode.")]
+#[clap(author, version, about)]
```
Comment (tnowacki) [inline language/move-compiler/src/bin/move-build.rs:14]: Could you inline the name/about like you did for `move-check`?

File: language/move-compiler/Cargo.toml [hunk 2]
```diff
@@ -2,6 +2,7 @@
 name = "move-compiler"
 version = "0.0.1"
 authors = ["Diem Association <opensource@diem.com>"]
+description = "Compile Move source to Move bytecode"
```
Comment (tnowacki) [inline language/move-compiler/Cargo.toml:5]: I think this actually a bit too specific for the compiler. You could remove it from this PR, or switch it to "The definition of the Move source language, and it's compiler" or something. There is a lot going on in here besides bytecode generation

**Review by tnowacki — COMMENTED** [review]
Thanks for the idiomatic fix! just a few small thing to change :)

**Comment by villesundell** [comment]
Thanks for the feedback :slightly_smiling_face: Suggestions implemented :+1: (I used the description you provided, since I think Cargo.toml should have one :nerd_face: )

**Comment by villesundell** [comment]
It seems `ci-test / Check for broken links in files (pull_request)` failed because https://opencollective.com/ was briefly offline, throwing `503` for each page.

**Comment by tnowacki** [comment]
@villesundell, sorry I missed this. Rerunning the tests now

**Comment by villesundell** [comment]
This time https://marketplace.visualstudio.com/manage/publishers/move threw 503 :nerd_face:

**Review by tnowacki — APPROVED** [review]

**Comment by tnowacki** [comment]
Thanks for the fix!

**Pull request merged**